modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/VersionEndpoint.java

```diff
@@ -0,0 +1,113 @@
+import org.opencastproject.util.doc.rest.RestService;
+
+import com.google.gson.Gson;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/")
+@RestService(name = "VersionService", title = "Version service",
+  abstractText = "Provides latest opencast version",
+  notes = { "This service offers the GET method to retrieve the latest opencast version from https://api.github.com ."})
+@Component(
+  immediate = true,
+  service = VersionEndpoint.class,
+  property = {
+    "service.description=Admin UI - Latest Version Endpoint",
+    "opencast.service.type=org.opencastproject.adminui.endpoint.VersionEndpoint",
+    "opencast.service.path=/admin-ng/oc-version"
+  }
+)
+public class VersionEndpoint {
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory.getLogger(VersionEndpoint.class);
+
+  /** The version */
+  private String version = "";
+
+  /** GitHub URL */
+  private String url = "https://api.github.com/repos/opencast/opencast/releases/latest";
+
+  /** The date */
+  private long lastUpdated = 0;
+
+  private static final Gson gson = new Gson();
+
+  /** OSGi callback. */
+  @Activate
+  protected void activate() {
+    logger.info("Activate the Admin ui - Latest version endpoint");
+  }
+
+  @GET
+  @Path("version.json")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "latestversion", description = "Returns the latest Opencast version",
+          returnDescription = "Returns the latest Opencast version retrieved from GitHub",
+          reponses = { @RestResponse(responseCode = SC_OK, description = "The latest Opencast version.") })
+  public String getVersion() {
+    if (System.currentTimeMillis() / 1000L - lastUpdated >= 3600) {
+      updateVersion();
+    }
+    return gson.toJson(version);
+  }
+
+  private synchronized void updateVersion() {
+    if (System.currentTimeMillis() / 1000L - lastUpdated < 3600) {
+      return;
+    }
+    try {
+      HttpClient client = HttpClientBuilder.create().build();
+      HttpGet request = new HttpGet(url);
+      HttpResponse response = client.execute(request);
+      String responseString = IOUtils.toString(response.getEntity().getContent(), "utf-8");
+      Map data = gson.fromJson(responseString, Map.class);
+      version = (String) data.get("tag_name");
+      lastUpdated = System.currentTimeMillis() / 1000L;
+    } catch (Exception e) {
+      logger.error("Error while parsing the version: " + e.toString());
```

Comment:
I would make this a warning since it's not something an admin would (or most likely even could) fix.
And use the logger's functionality to handle exceptions:

```suggestion
      logger.error("Error while parsing the Opencast version from GitHub", e);
```

Reply:
How do I make this a warning? 
Does the logger.error() function handle the Exception in a way such that it will create a warning?

Reply:
`logger.warn(…)`